aten/src/ATen/native/UnaryOps.cpp

```diff
@@ -153,14 +153,28 @@ TORCH_IMPL_FUNC(func_out) (const Tensor& self, const Tensor& result) {  \
   func_stub(device_type(), *this);                                      \
 }
 
+// This macro is as optional as the one above. torch.(ceil|floor|round|trunc) are no-ops for integers
+// See gh-70918
+#define CREATE_UNARY_TORCH_IMPL_INTEGER_NO_OP_FUNC(func_out, func_stub)                                \
+TORCH_IMPL_FUNC(func_out) (const Tensor& self, const Tensor& result) {  \
+  if (c10::isIntegralType(self.scalar_type(), /*includeBool=*/false)) {                                      \
+    result.copy_(self);                                                 \
```

Comment:
Wait... shouldn't we be returning self in these cases?

Reply:
array api doesn't say that, numpy currently even returns different data type than input (that diverges from array api):
```
In [59]: out=np.ceil(anp)

In [60]: out
Out[60]: array([3., 0., 1., 3.])

In [61]: anp
Out[61]: array([3, 0, 1, 3])
```

Reply:
@ngimel and @mruberry does this look good to go then?

Reply:
@mruberry does this need changes?

Reply:
It'd be good to add a small note in the docs, similar to that in the array API, noting this special case and what we do (is it a view of the input? is it a fresh new tensor?)

Reply:
Makes sense, I'll add to the docs of the changed functions. We are currently returning a copy of the tensor. For integer inputs.